specs/Standards/MultiToken/Metadata.md

```diff
@@ -0,0 +1,141 @@
+
+Prior art:
+
+- NEAR's [Fungible Token Metadata Standard](../FungibleToken/Metadata.md)
+- NEAR's [Non-Fungible Token Metadata Standard](../NonFungibleToken/Metadata.md)
+- Discussion about NEAR's complete NFT standard: #171
+- Discussion about NEAR's complete Multi Token standard: #246
+
+## Interface
+
+Metadata applies at both the class level (`MTBaseTokenMetadata`) and the specific instance level (`MTTokenMetadata`). The relevant metadata for each:
+
+```ts
+
+type MTContractMetadata = {
+  spec: string, // required, essentially a version like "mt-1.0.0"
+  name: string, // required Zoink's Digitial Sword Collection
+}
+
+type MTBaseTokenMetadata = {
+  name: string, // required, ex. "Silver Swords" or "Metaverse 3"
+  id: string, // required a unique identifier for the metadata
+  symbol: string|null, // required, ex. "MOCHI"
+  icon: string|null, // Data URL
+  decimals: string|null // number of decimals for the token useful for FT related tokens
+  base_uri: string|null, // Centralized gateway known to have reliable access to decentralized storage assets referenced by `reference` or `media` URLs
+  reference: string|null, // URL to a JSON file with more info
+  copies: number|null, // number of copies of this set of metadata in existence when token was minted.
+  reference_hash: string|null, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadata = {
+  title: string|null, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
+  description: string|null, // free-form description
+  media: string|null, // URL to associated media, preferably to decentralized, content-addressed storage
+  media_hash: string|null, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+  issued_at: string|null, // When token was issued or minted, Unix epoch in milliseconds
+  expires_at: string|null, // When token expires, Unix epoch in milliseconds
+  starts_at: string|null, // When token starts being valid, Unix epoch in milliseconds
+  updated_at: string|null, // When token was last updated, Unix epoch in milliseconds
+  extra: string|null, // Anything extra the MT wants to store on-chain. Can be stringified JSON.
+  reference: string|null, // URL to an off-chain JSON file with more info.
+  reference_hash: string|null // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadataAll = {
+  base: MTBaseTokenMetadata
+  token: MTTokenMetadata
+}
+```
+
+A new set of functions MUST be supported on the MT contract:
+
+```ts
+// Returns the top-level contract level metadtata
+function mt_metadata_contract(): MTContractMetadata {}
+function mt_metadata_token_all(token_ids: string[]): MTTokenMetadataAll[]
+function mt_metadata_token_by_token_id(token_ids: string[]): MTTokenMetadata[]
+function mt_metadata_base_by_token_id(token_ids: string[]): MTBaseMetadata[]
```

Comment:
What is the MTBaseMetadata struct? I can't seem to find it. Should this be the MTBaseTokenMetadata?

Reply:
yes, nice catch!